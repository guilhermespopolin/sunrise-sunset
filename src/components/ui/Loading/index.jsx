import React from 'react'
import styled, { keyframes } from 'styled-components'

import Box from 'components/ui/Box'

const upDown = keyframes`
  0% {
    transform: translate(0, 0);
  }

  50% {
    transform: translate(0, 15px);
  }

  100% {
    transform: translate(0, 0);
  }
`

const Ball = styled.div`
  display: inline-block;
  width: 15px;
  height: 15px;
  border-radius: 15px;
  background-color: ${({ theme }) => theme.colors.secondary};

  &[data-id="1"] {
    animation: ${upDown} 0.6s 0.1s linear infinite;
  }

  &[data-id="2"] {
    animation: ${upDown} 0.6s 0.2s linear infinite;
  }

  &[data-id="3"] {
    animation: ${upDown} 0.6s 0.3s linear infinite;
  }

  &:not(:last-child) {
    margin-right: 0.5rem;
  }
`

function Loading() {
  return (
    <Box as="div" data-testid="loading">
      <Ball data-id={1} />
      <Ball data-id={2} />
      <Ball data-id={3} />
    </Box>
  )
}

export default Loading
