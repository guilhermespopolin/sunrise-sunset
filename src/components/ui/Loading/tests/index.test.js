import React from 'react'
import { render } from '__tests__/utils/rtlWrapper'

import Loading from '..'

describe('<Loading />', () => {
  it('should render properly', () => {
    const { getByTestId } = render(<Loading />)

    expect(getByTestId('loading')).toMatchSnapshot()
  })
})

