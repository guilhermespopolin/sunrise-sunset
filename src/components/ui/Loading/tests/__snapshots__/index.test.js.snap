// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loading /> should render properly 1`] = `
.c0 {
  display: inline-block;
  width: 15px;
  height: 15px;
  border-radius: 15px;
  background-color: #ff9900;
}

.c0[data-id="1"] {
  -webkit-animation: ioOCvU 0.6s 0.1s linear infinite;
  animation: ioOCvU 0.6s 0.1s linear infinite;
}

.c0[data-id="2"] {
  -webkit-animation: ioOCvU 0.6s 0.2s linear infinite;
  animation: ioOCvU 0.6s 0.2s linear infinite;
}

.c0[data-id="3"] {
  -webkit-animation: ioOCvU 0.6s 0.3s linear infinite;
  animation: ioOCvU 0.6s 0.3s linear infinite;
}

.c0:not(:last-child) {
  margin-right: 0.5rem;
}

<div
  class=""
  data-testid="loading"
>
  <div
    class="c0"
    data-id="1"
  />
  <div
    class="c0"
    data-id="2"
  />
  <div
    class="c0"
    data-id="3"
  />
</div>
`;
